src/category_theory/monoidal/internal/types.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.category.Mon.basic
+import category_theory.monoidal.internal
+import category_theory.monoidal.types
+
+/-!
+# `Mon_ (Type u) ≌ Mon.{u}`
+
+The category of internal monoid objects in `Type`
+is equivalent to the category of "native" bundled monoids.
+
+Moreover, this equivalence is compatible with the forgetful functors to `Type`.
+-/
+
+universes v u
+
+open category_theory
+open category_theory.monoidal
+
+namespace Mon_Type_equivalence_Mon
+
+/--
+Converting a monoid object in `Type` to a bundled monoid.
+-/
+def functor : Mon_ (Type u) ⥤ Mon.{u} :=
+{ obj := λ A, ⟨A.X,
+  { one := A.one punit.star,
+    mul := λ x y, A.mul (x, y),
+    one_mul := λ x, by convert congr_fun A.one_mul (punit.star, x),
+    mul_one := λ x, by convert congr_fun A.mul_one (x, punit.star),
+    mul_assoc := λ x y z, by convert congr_fun A.mul_assoc ((x, y), z), }⟩,
+  map := λ A B f,
+  { to_fun := f.hom,
+    map_one' := congr_fun f.one_hom punit.star,
+    map_mul' := λ x y, congr_fun f.mul_hom (x, y), }, }.
```

Comment:
Are these `.` at the end of the definitions stylistic? Or are they evidence that this code is somewhat slowish?

Reply:
It's not exactly snappy, but compiles happily with `-T40000`. `inverse` is the culprit, taking 8s to elaborate. `tidy` is doing a lot of work here.